src/node/rpc/node_frontend.h

```diff
@@ -1001,6 +1001,48 @@ namespace ccf
         .set_execute_outside_consensus(
           ccf::endpoints::ExecuteOutsideConsensus::Locally)
         .install();
+
+      auto update_identity = [this](auto& args, const nlohmann::json& params) {
+        const auto in = params.get<UpdateIdentity::In>();
+        auto ids = args.tx.rw(network.byzantine_network_identities);
+        bool exists = false;
+        ids->foreach(
+          [rid = in.rid, &exists](
+            const kv::ReconfigurationId& nid, const ByzantineIdentity& id) {
+            if (nid == rid)
+            {
+              exists = true;
+              return false;
+            }
+            return true;
+          });
+
+        if (exists)
+        {
+          return make_error(
+            HTTP_STATUS_BAD_REQUEST,
+            ccf::errors::IdentityAlreadyExists,
+            fmt::format("identity for configuration {} exists", in.rid));
+        }
+
+        // For now, just pretend that we're done and that we have a shared
+        // identity.
+        ByzantineIdentity byid;
+        byid.cert = network.identity->cert;
+        ids->put(in.rid, byid);
+        return make_success(true);
+      };
+
+      make_endpoint(
+        "update-identity",
```

Comment:
should this be called create-identity? it seems like this will fail if the identity exists

Reply:
This is where the Byzantine identity code will go; it will make many calls to this endpoint before the identity is actually established, but today we don't have that yet, so this RPC will immediately create a copy of the existing identity.

Reply:
```suggestion
        "update_identity",
```

For consistency with existing endpoint naming style.